cloudinit/sources/DataSourceSoftLayer.py

```diff
@@ -0,0 +1,63 @@
+# vi: ts=4 expandtab
+#
+#    Copyright (C) 2012 Canonical Ltd.
+#    Copyright (C) 2012 Yahoo! Inc.
+#
+#    Author: Peter Schroeter <schroeter@gmail.com>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License version 3, as
+#    published by the Free Software Foundation.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+
+from cloudinit import log as logging
+from cloudinit import sources
+from cloudinit import util
+from cloudinit import url_helper
+ 
+LOG = logging.getLogger(__name__)
+SOFTLAYER_API_QUERY_URL = 'https://api.service.softlayer.com/rest/v3.1/SoftLayer_Resource_Metadata'
+
+class DataSourceSoftLayer(sources.DataSource):
+    def __init__(self, sys_cfg, distro, paths):
+        sources.DataSource.__init__(self, sys_cfg, distro, paths)
+        self.seed_dir = os.path.join(paths.seed_dir, 'sl')
+        self.metadata = {}
+        self.userdata_raw = None
+
+    def _fetch_metadata_item(self, item):
+        item_url = url_helper.combine_url(SOFTLAYER_API_QUERY_URL, item)
```

Comment:
cloud-init wasn't working for SL with RLL boot, this combine_url function doesn't exist in this version of cloud-init